### PR TITLE
Support Limit and Offset as part of the opts map

### PR DESCRIPTION
**Pull request by StareIntoTheBeard**

### CODE REVIEW

File: lib/jsonapi/query_parser.ex [hunk 1]
```diff
@@ -65,6 +65,8 @@ defmodule JSONAPI.QueryParser do
     |> parse_include(Map.get(query_params, "include", ""))
     |> parse_filter(Map.get(query_params, "filter", %{}))
     |> parse_sort(Map.get(query_params, "sort", ""))
+    |> Map.put(:limit, Map.get(query_params, "limit", ""))
```
Comment (doomspork) [inline lib/jsonapi/query_parser.ex]: If we're going to add these, we should add all the query params JSON API supports for pagination:

+ limit
+ offset
+ size
+ page
+ cursor

These should probably return integers and not strings with `nil` being a default and not `""`, don't you think?

We can probably refactor this whole pipe when we do add the aforementioned params to be a big cleaner and avoid the nested function calls.

Comment (StareIntoTheBeard) [reply lib/jsonapi/query_parser.ex]: Yeah, I'll buy that -- side note is there a native ecto syntax for `cursor`?

Comment (StareIntoTheBeard) [reply lib/jsonapi/query_parser.ex]: Main question being, what values are we expecting that to come across with?

**Comment by jeregrine** [comment]
@StareIntoTheBeard @doomspork SO the jsonapi spec is very clear on how its agnostic about paging. It says that the "page"keyword is reserved for paging and its to be used like `page[PAGING_TYPE]` http://jsonapi.org/format/#fetching-pagination so if anything I'd support adding the "page" keyword and nothing else.

I'd mostly suggest just grabbing the page keyword from the regular query params and handling it yourself.

I do appreciate the PR though. If you have anything else you'd like to see lemme know.

**Comment by StareIntoTheBeard** [comment]
@jeregrine The point was I wanted it in the same struct with the rest of the JSON API params. I could have easily handed down more query params to Ecto but the spec does handle pagination (in the way you've said, thanks for pointing that out) but I'd like it in the same assigns struct... Made some changes and will reopen and welcome comments.

**Pull request closed without merging**